### PR TITLE
Style improvements

### PR DESCRIPTION
**Pull request by flovilmart**
- Enable HooksRouter by default (as part of the node SDK)
- Moves problematic middlewares on per route basis
- Modernizes DatabaseController and middleware imports
- Moves test api mount to helper

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
@flovilmart updated the pull request.

**Comment by facebook-github-bot-4** [comment]
@flovilmart updated the pull request.

File: src/Controllers/AnalyticsController.js [hunk 1]
```diff
@@ -3,21 +3,23 @@ import { AnalyticsAdapter } from '../Adapters/Analytics/AnalyticsAdapter';
...
 export class AnalyticsController extends AdaptableController {
   appOpened(req) {
-    return this.adapter.appOpened(req.body, req).then(
-      function(response) {
-        return { response: response };
-      }).catch((err) => {
-        return { response: {} };
-      });
+    return Promise.resolve().then(() => {
+      this.adapter.appOpened(req.body, req);
```
Comment (steven-supersolid) [inline src/Controllers/AnalyticsController.js]: Missing return?

File: src/PromiseRouter.js [hunk 2]
```diff
@@ -107,47 +112,17 @@ export default class PromiseRouter {
...
-        expressApp.put(route.path, makeExpressHandler(this.appId, route.handler));
-        break;
-      case 'DELETE':
-        expressApp.delete(route.path, makeExpressHandler(this.appId, route.handler));
-        break;
-      default:
-        throw 'unexpected code branch';
-      }
-    }
+    this.routes.forEach((route) => {
```
Comment (steven-supersolid) [inline src/PromiseRouter.js]: Assume safe to add methods that were previously unsupported?

File: src/Controllers/AnalyticsController.js (hunk 1 above)
Comment (flovilmart) [inline src/Controllers/AnalyticsController.js]: Yup

File: src/PromiseRouter.js (hunk 2 above)
Comment (flovilmart) [inline src/PromiseRouter.js]: It's not unsafe and totally internal.

File: src/index.js [hunk 3]
```diff
@@ -17,4 +16,4 @@ _ParseServer.createLiveQueryServer = ParseServer.createLiveQueryServer;
 let GCSAdapter = useExternal('GCSAdapter', 'parse-server-gcs-adapter');
 
 export default ParseServer;
-export { S3Adapter, GCSAdapter, FileSystemAdapter, InMemoryCacheAdapter, TestUtils, logger, _ParseServer as ParseServer };
+export { S3Adapter, GCSAdapter, FileSystemAdapter, InMemoryCacheAdapter, logger, _ParseServer as ParseServer };
```
Comment (drew-gross) [inline src/index.js]: JS SDK uses `TestUtils` https://github.com/ParsePlatform/Parse-SDK-JS/blob/a2b55078d991feada4fb4497625d79be63e208c4/integration/server.js

**Comment by drew-gross** [comment]
Looks awesome, just need to either keep TestUtils exported, or change the JS SDK. Feel free to merge after that.

**Comment by flovilmart** [comment]
I believe it's good now, let me know if the TestUtils change is good :)

**Pull request merged**